algolia-search/src/search-form.tsx

```diff
@@ -0,0 +1,158 @@
+import {
+  useCurrentRefinements,
+  useInstantSearch,
+  useRefinementList,
+  useSearchBox
+} from "react-instantsearch";
+import {useEffect, useRef} from "preact/compat";
+
+const SearchForm = (props) => {
+
+  const {query, refine} = useSearchBox(props);
+  const {items: pageTypeRefinements, refine: refinePageTypes} = useRefinementList({attribute: "basic_page_type"});
+  const {items: sharedRefinements, refine: refineSharedTypes} = useRefinementList({attribute: "shared_tags"});
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(window.location.search);
+    searchParams.delete("key")
+    searchParams.delete("page-type")
+    searchParams.delete("shared")
+
+    if (query) searchParams.set("key", query);
+    const pageTypes = pageTypeRefinements.filter(item => item.isRefined).map(item => item.value);
+    const sharedTypes = sharedRefinements.filter(item => item.isRefined).map(item => item.value);
+
+    if (pageTypes.length >= 1) searchParams.set("page-type", pageTypes.join(','))
+    if (sharedTypes.length >= 1) searchParams.set("shared-type", sharedTypes.join(','))
```

Comment:
Needs to match your GET parameter: https://github.com/SU-SWS/c_sharp_subtheme/pull/7/files#diff-0e8857431b6bb5e1b1e94ffc0c3418f5dcf66c9ae354b8efac92cd15613bb551R83